ts/screens/authentication/IdpLoginScreen.tsx

```diff
@@ -26,14 +26,24 @@ type OwnProps = {
 
 type Props = ReturnType<typeof mapStateToProps> & ReduxProps & OwnProps;
 
+/**
+ * web request can be viewed in one of these states:
+ * 'loading' is the initial state, the webview receives the url directly from props so it starts to load
+ * 'completed' means the request has been successfully executed
+ * 'error' means the request got an error (endpoint unreachable, timeout, http errors (404,500...))
+ */
+type RequestState = "LOADING" | "COMPLETED" | "ERROR";
```

Comment:
you can probably use a `Pot` here as it covers all these states: https://github.com/teamdigitale/io-ts-commons/blob/master/src/pot.ts